Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -725,16 +725,24 @@ class AnimatedValue extends AnimatedWithChildren {
    * things like the start of a pan gesture.
    */
   setOffset(offset: number): void {
-    this._offset = offset;
+    if (this.__isNative) {
+      NativeAnimatedAPI.setAnimatedNodeOffset(this.__getNativeTag(), offset);
+    } else {
+      this._offset = offset;
+    }
   }
 
   /**
    * Merges the offset value into the base value and resets the offset to zero.
    * The final output of the value is unchanged.
    */
   flattenOffset(): void {
-    this._value += this._offset;
-    this._offset = 0;
+    if (this.__isNative) {
+      NativeAnimatedAPI.flattenAnimatedNodeOffset(this.__getNativeTag());
+    } else {
```

Comment:
Same here, we should try to keep the JS values in sync even if it is native driven.